Windows/scratch-link/BLESession.cs

```diff
@@ -301,6 +301,14 @@ private async Task Connect(JObject parameters)
             _peripheral.ConnectionStatusChanged += OnPeripheralStatusChanged;
             _services = servicesResult.Services;
 
+            // cache all characteristics in all services
+            foreach (var service in _services) {
+                var characteristicsResult = await service.GetCharacteristicsAsync(BluetoothCacheMode.Uncached);
```

Comment:
I think we should handle the possibility of `characteristicsResult.Status` being something other than `Success`. Maybe something like the code below. Do you agree, or perhaps have another suggestion?

```suggestion
                var characteristicsResult = await service.GetCharacteristicsAsync(BluetoothCacheMode.Uncached);
                if (characteristicsResult.Status != GattCommunicationStatus.Success) {
                    continue; // we'll try again later if the client tries to access this characteristic
                }
```

Reply:
That is a good point. I fully agree with you.
I have not encountered any failed cases yet so far, but your code is necessary to avoid caching incorrect data.

I am going to add the code shortly. Thanks!

Reply:
Although only success cases, the updated code has been verified with micro:bit and LEGO BOOST as well as our device.